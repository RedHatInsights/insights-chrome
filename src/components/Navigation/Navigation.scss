.chr-c-app-title {
  color: white;
  padding: var(--pf-v5-global--spacer--md) var(--pf-v5-global--spacer--sm) var(--pf-v5-global--spacer--md) var(--pf-v5-global--spacer--lg);
  width: 100%;
  border-bottom: var(--pf-v5-global--spacer--xs) solid var(--pf-v5-global--Color--300);
  font-weight: var(--pf-v5-global--FontWeight--semi-bold);
  white-space: normal;
  text-align: left;
  font-size: 17px;
}

.pf-v6-c-nav__list {
  padding-top: 0;
  section.chr-c-section-nav {
    margin-bottom: var(--pf-v5-global--spacer--md);
    margin-top: 0 !important;
    h2.pf-v6-c-nav__section-title {
      border: 0;
      border-top: var(--pf-v5-global--BorderWidth--sm) var(--pf-v5-global--BackgroundColor--dark-200) solid;
      color: var(--pf-v5-global--palette--black-500);
      font-size: var(--pf-v5-global--FontSize--xs);
      font-weight: var(--pf-v5-global--FontWeight--medium);
      margin-bottom: 0;
      padding: var(--pf-v5-global--spacer--md) 0 var(--pf-v5-global--spacer--lg) var(--pf-v5-global--spacer--lg);
    }
    ul
      > li.pf-v6-c-nav__item {
        border-top: 0;
        --pf-v5-c-nav__item--MarginTop: 0;
        --pf-v5-c-nav__item--before--BorderWidth: 0;
        &:last-child {
          margin-bottom: var(--pf-v5-global--spacer--sm);
        }
        .pf-v6-c-nav__subnav {
          padding: 0 0 0 var(--pf-v5-global--spacer--lg);
          .pf-v6-c-nav__list {
            .pf-v6-c-nav__item {
              .pf-v6-c-nav__link{
                padding-left: var(--pf-v5-global--spacer--lg);
              }
              &:last-child {
                margin-bottom: var(--pf-v5-global--spacer--sm);
              }
            }
          }
        }
        section.pf-v6-c-nav__subnav {
          li.pf-v6-c-nav__item {
            border: none !important;
          }
        }
      }
    }
    .pf-v6-c-nav__item.chr-c-navigation__additional-links {
      .pf-v6-c-nav__link {
        padding-left: var(--pf-v5-global--spacer--lg);
      }
    }
  > .pf-v6-c-nav__item:not(:first-child) {
    border-top: var(--pf-v5-global--BorderWidth--sm) var(--pf-v5-global--BackgroundColor--dark-200) solid;
  }   
  svg {
    margin-left: var(--pf-v5-global--spacer--sm);
    width: 12px;  
  }
}

.pf-v6-c-nav__item {
  a.chr-c-navigation__with-notifier {
    align-items: center;
    .notifier-icon {
      margin-left: auto;
      margin-right: 8px;
    }
  }
}

.pf-v6-c-nav {
  .chr-c-navigation__beta-icon {
    --pf-v5-c-icon__content--Color: var(--pf-v5-global--palette--green-300);
  }
}

.chr-c-navigation{
  &__isHidden {
    display: none;
  }
  &__item {
    align-items: center;
  }
}

.chr-c-app-switcher--loading {
  padding: var(--pf-v5-global--spacer--md) var(--pf-v5-global--spacer--lg)
}