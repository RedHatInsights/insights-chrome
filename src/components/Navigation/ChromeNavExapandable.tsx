import React from 'react';
import { NavExpandable } from '@patternfly/react-core';
import ChromeNavItemFactory from './ChromeNavItemFactory';
import { useSelector } from 'react-redux';
import { isFedRamp } from '../../utils/common';
import { computeFedrampResult } from '../../utils/useRenderFedramp';
import { ReduxState, RouteDefinition } from '../../redux/store';

export type ChromeNavExapandableProps = {
  title: string;
  routes: RouteDefinition[];
  active?: boolean;
  isHidden?: boolean;
  id?: string;
};

const ChromeNavExapandable = ({ title, routes, active, isHidden, id }: ChromeNavExapandableProps) => {
  const modules = useSelector((state: ReduxState) => state.chrome.modules);
  let filteredFedrampRoutes = routes;
  if (isFedRamp()) {
    filteredFedrampRoutes = routes.filter(({ appId, href }) => {
      return appId && computeFedrampResult(appId, href, modules![appId]);
    });
  }

  if (isHidden || filteredFedrampRoutes.filter(({ appId }) => !!appId).length === 0) {
    return null;
  }

  const quickStartHighlightId = title.replace(/\s/g, '-');
  return (
    <NavExpandable id={id} isExpanded={active} isActive={active} title={title} data-quickstart-id={quickStartHighlightId}>
      {filteredFedrampRoutes.map((item, index) => (
        <ChromeNavItemFactory key={index} {...item} />
      ))}
    </NavExpandable>
  );
};

export default ChromeNavExapandable;
