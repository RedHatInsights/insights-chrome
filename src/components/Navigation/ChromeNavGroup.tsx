import React from 'react';
import PropTypes from 'prop-types';
import { NavGroup } from '@patternfly/react-core';

import WrenchIcon from '@patternfly/react-icons/dist/js/icons/wrench-icon';
import SecurityIcon from '@patternfly/react-icons/dist/js/icons/security-icon';
import TrendUpIcon from '@patternfly/react-icons/dist/js/icons/trend-up-icon';
import CodeIcon from '@patternfly/react-icons/dist/js/icons/code-icon';
import DatabaseIcon from '@patternfly/react-icons/dist/js/icons/database-icon';
import CloudIcon from '@patternfly/react-icons/dist/js/icons/cloud-upload-alt-icon';

import ChromeNavItemFactory from './ChromeNavItemFactory';
import { NavItem } from '../../@types/types';

const sectionTitleMapper = {
  wrench: <WrenchIcon />,
  shield: <SecurityIcon />,
  database: <DatabaseIcon />,
  cloud: <CloudIcon />,
  code: <CodeIcon />,
  'trend-up': <TrendUpIcon />,
};

export type ChromeNavGroupProps = {
  navItems: NavItem[];
  isHidden?: boolean;
  icon?: keyof typeof sectionTitleMapper;
  title: string;
};

const ChromeNavGroup = ({ navItems, isHidden, icon, title }: ChromeNavGroupProps) => {
  const filteredFedrampNavItems = navItems;

  if (isHidden) {
    return null;
  }

  const groupTitle = (
    <div>
      {icon && sectionTitleMapper[icon]}
      {title}
    </div>
  );
  return (
    // PF does not allow node/element in the NavGroup prop types, hence the type cast
    <NavGroup className="chr-c-section-nav" id={title} title={groupTitle as unknown as string}>
      {filteredFedrampNavItems.map((item, index) => (
        <ChromeNavItemFactory key={index} {...item} />
      ))}
    </NavGroup>
  );
};

ChromeNavGroup.propTypes = {
  navItems: PropTypes.array.isRequired,
  icon: PropTypes.oneOf(['wrench', 'shield', 'trend-up', 'database', 'cloud', 'code']),
  title: PropTypes.string.isRequired,
  isHidden: PropTypes.bool,
};

export default ChromeNavGroup;
