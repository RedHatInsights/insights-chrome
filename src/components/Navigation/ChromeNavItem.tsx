import React, { useEffect, useMemo } from 'react';
import { Icon } from '@patternfly/react-core/dist/dynamic/components/Icon';
import { NavItem } from '@patternfly/react-core/dist/dynamic/components/Nav';
import { Tooltip } from '@patternfly/react-core/dist/dynamic/components/Tooltip';
import ExternalLinkAltIcon from '@patternfly/react-icons/dist/dynamic/icons/external-link-alt-icon';
import FlaskIcon from '@patternfly/react-icons/dist/dynamic/icons/flask-icon';
import BellIcon from '@patternfly/react-icons/dist/dynamic/icons/bell-icon';
import StarIcon from '@patternfly/react-icons/dist/dynamic/icons/star-icon';
import { titleCase } from 'title-case';
import classNames from 'classnames';
import get from 'lodash/get';
import { useAtomValue, useSetAtom } from 'jotai';

import ChromeLink, { LinkWrapperProps } from '../ChromeLink/ChromeLink';
import { useSelector } from 'react-redux';
import { ChromeNavItemProps } from '../../@types/types';
import useFavoritePagesWrapper from '../../hooks/useFavoritePagesWrapper';
import { isPreviewAtom } from '../../state/atoms/releaseAtom';
import { activeProductAtom } from '../../state/atoms/activeProductAtom';

const ChromeNavItem = ({
  appId,
  className,
  href,
  isHidden,
  ignoreCase,
  title,
  isExternal,
  isBeta: isBetaEnv,
  active,
  product,
  notifier = '',
}: ChromeNavItemProps) => {
  const isPreview = useAtomValue(isPreviewAtom);
  const hasNotifier = useSelector((state) => get(state, notifier));
  const markActiveProduct = useSetAtom(activeProductAtom);
  const { favoritePages } = useFavoritePagesWrapper();
  const isFavorited = useMemo(() => favoritePages.find(({ favorite, pathname }) => favorite && pathname === href), [href, favoritePages]);

  useEffect(() => {
    if (active) {
      markActiveProduct(product);
    }
  }, [active]);

  if (isHidden) {
    return null;
  }

  return (
    <NavItem
      className={classNames('chr-c-navigation__item', className, { 'chr-c-navigation__with-notifier': hasNotifier })}
      itemID={href}
      data-quickstart-id={href}
      preventDefault
      isActive={active}
      to={href}
      ouiaId={title}
      component={(props: LinkWrapperProps) => <ChromeLink {...props} isBeta={isBetaEnv} isExternal={isExternal} appId={appId} />}
    >
      {typeof title === 'string' && !ignoreCase ? titleCase(title) : title}{' '}
      {isExternal && (
        <Icon isInline>
          <ExternalLinkAltIcon />
        </Icon>
      )}
      {isBetaEnv && !isPreview && !isExternal && (
        <Tooltip position={'right'} content={<div>This service is a Preview.</div>}>
          <Icon className="chr-c-navigation__beta-icon" isInline>
            <FlaskIcon />
          </Icon>
        </Tooltip>
      )}
      {isFavorited && (
        <Icon>
          <StarIcon color="var(--pf-t--temp--dev--tbd)"/* CODEMODS: original v5 color was --pf-v5-global--palette--gold-400 */ />
        </Icon>
      )}
      {hasNotifier && (
        <Icon size="md">
          <BellIcon className="notifier-icon" color="var(--pf-v5-global--default-color--200)" />
        </Icon>
      )}
    </NavItem>
  );
};

export default ChromeNavItem;
