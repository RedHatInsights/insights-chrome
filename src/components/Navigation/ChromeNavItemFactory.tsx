import React, { useContext } from 'react';
import { ChromeNavExapandableProps } from './ChromeNavExapandable';
import { ChromeNavGroupProps } from './ChromeNavGroup';
import { ChromeNavItemProps } from './ChromeNavItem';
import { DynamicNavProps } from './DynamicNav';
import NavContext from './navContext';

export type ChromeNavItemFactoryProps = {
  groupId?: string;
  expandable?: boolean;
  dynamicNav?: string;
};

const isNavExpandableProps = (props: Record<string, any>, expandable?: boolean): props is ChromeNavExapandableProps => expandable === true;
const isNavGroupProps = (props: Record<string, any>, groupId?: string): props is ChromeNavGroupProps => typeof groupId === 'string';
const isDynamicNavProps = (props: Record<string, any>, dynamicNav?: string): props is DynamicNavProps => typeof dynamicNav === 'string';

const ChromeNavItemFactory = ({ groupId, expandable, dynamicNav, ...rest }: ChromeNavItemFactoryProps) => {
  const { componentMapper } = useContext(NavContext);
  const props = rest;
  if (isNavGroupProps(props, groupId)) {
    return <componentMapper.group {...props} />;
  } else if (isNavExpandableProps(props, expandable)) {
    return <componentMapper.expandable {...props} />;
  } else if (isDynamicNavProps(props, dynamicNav)) {
    return <componentMapper.dynamicNav {...props} dynamicNav={dynamicNav!} />;
  } else {
    return <componentMapper.item {...(props as ChromeNavItemProps)} />;
  }
};

export default ChromeNavItemFactory;
