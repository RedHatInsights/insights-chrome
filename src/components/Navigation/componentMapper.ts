import ChromeNavGroup from './ChromeNavGroup';
import ChromeNavExapandable from './ChromeNavExapandable';
import ChromeNavItem from './ChromeNavItem';
import DynamicNav from './DynamicNav';

const componentMapper = {
  group: ChromeNavGroup,
  expandable: ChromeNavExapandable,
  item: ChromeNavItem,
  dynamicNav: DynamicNav,
};

export default componentMapper;
