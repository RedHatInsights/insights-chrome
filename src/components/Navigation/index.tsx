import React, { Fragment, useRef, useState } from 'react';
import { Nav, NavList } from '@patternfly/react-core/dist/dynamic/components/Nav';
import { PageContextConsumer } from '@patternfly/react-core/dist/dynamic/components/Page';
import { useAtomValue } from 'jotai';

import NavContext from './navContext';
import componentMapper from './componentMapper';
import ChromeNavItemFactory from './ChromeNavItemFactory';
import BetaInfoModal from '../../components/BetaInfoModal';

import NavLoader from './Loader';
import ChromeNavItem from './ChromeNavItem';
import type { Navigation as NavigationSchema } from '../../@types/types';
import { useFlag } from '@unleash/proxy-client-react';
import { getUrl } from '../../hooks/useBundle';
import { isPreviewAtom } from '../../state/atoms/releaseAtom';

export type NavigationProps = { loaded: boolean; schema: NavigationSchema };

const Navigation: React.FC<NavigationProps> = ({ loaded, schema }) => {
  const isPreview = useAtomValue(isPreviewAtom);
  const [showBetaModal, setShowBetaModal] = useState(false);
  const deferedOnClickArgs = useRef<[React.MouseEvent<HTMLAnchorElement, MouseEvent> | undefined, string | undefined, string | undefined]>([
    undefined,
    undefined,
    undefined,
  ]);
  const showBundleCatalog = localStorage.getItem('chrome:experimental:quickstarts') === 'true';
  const breadcrumbsDisabled = !useFlag('platform.chrome.bredcrumbs.enabled');

  const onLinkClick = (origEvent: React.MouseEvent<HTMLAnchorElement, MouseEvent>, href: string) => {
    if (!showBetaModal && !isPreview) {
      origEvent.preventDefault();
      deferedOnClickArgs.current = [origEvent, href, origEvent?.currentTarget?.text];
      setShowBetaModal(true);
      return false;
    }

    return true;
  };

  if (!loaded) {
    return <NavLoader />;
  }

  return (
    <Fragment>
      {breadcrumbsDisabled && <div className="chr-c-app-title">{schema?.title}</div>}
      <Nav aria-label="Insights Global Navigation" data-ouia-safe="true" ouiaId="SideNavigation">
        <NavList>
          <PageContextConsumer>
            {({ isSidebarOpen }) => (
              <NavContext.Provider
                value={{
                  componentMapper,
                  onLinkClick,
                  inPageLayout: true,
                  isNavOpen: isSidebarOpen,
                }}
              >
                {schema.navItems.map((item, index) => (
                  <ChromeNavItemFactory key={index} {...item} />
                ))}
                {showBundleCatalog ? <ChromeNavItem title="Quickstarts" href={`/${getUrl('bundle')}/quickstarts`} appId="dynamic" /> : <Fragment />}
              </NavContext.Provider>
            )}
          </PageContextConsumer>
        </NavList>
      </Nav>
      <BetaInfoModal
        isOpen={showBetaModal}
        onClick={(event) => {
          if (!isPreview) {
            const [origEvent, href] = deferedOnClickArgs.current;
            const isMetaKey = event.ctrlKey || event.metaKey || origEvent?.ctrlKey || origEvent?.metaKey;
            const url = `${document.baseURI}beta${href}`;
            isMetaKey ? window.open(url) : (window.location.href = url);
          }
        }}
        onCancel={() => setShowBetaModal(false)}
        menuItemClicked={deferedOnClickArgs.current[2]}
      />
    </Fragment>
  );
};

export default Navigation;
