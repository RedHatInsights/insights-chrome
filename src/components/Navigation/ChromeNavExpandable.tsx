import React from 'react';
import { NavExpandable } from '@patternfly/react-core/dist/dynamic/components/Nav';
import ChromeNavItemFactory from './ChromeNavItemFactory';
import { ChromeNavExpandableProps } from '../../@types/types';

const ChromeNavExpandable = ({ title, routes, active, isHidden, id }: ChromeNavExpandableProps) => {
  if (isHidden || routes.filter(({ appId, expandable }) => expandable || !!appId).length === 0) {
    return null;
  }

  const quickStartHighlightId = title.replace(/\s/g, '-');
  return (
    <NavExpandable id={id} isExpanded={active} isActive={active} title={title} data-quickstart-id={quickStartHighlightId}>
      {routes.map((item, index) => (
        <ChromeNavItemFactory key={index} {...item} />
      ))}
    </NavExpandable>
  );
};

export default ChromeNavExpandable;
