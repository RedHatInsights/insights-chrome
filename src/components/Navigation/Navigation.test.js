/* eslint-disable react/prop-types */
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { act, render } from '@testing-library/react';
import createMockStore from 'redux-mock-store';
import Navigation from './';

jest.mock('axios', () => {
  const axios = jest.requireActual('axios');
  return {
    __esModule: true,
    ...axios,
    get: () =>
      Promise.resolve({
        data: {
          navItems: [],
        },
      }),
    default: {
      ...axios.default,
      get: () =>
        Promise.resolve({
          data: {
            navItems: [],
          },
        }),
    },
  };
});

import * as axios from 'axios';

jest.mock('@unleash/proxy-client-react', () => {
  const actual = jest.requireActual('@unleash/proxy-client-react');
  return {
    __esModule: true,
    ...actual,
    // unblock navigation loading
    useFlagsStatus: () => ({ flagsReady: true }),
  };
});

const NavContextWrapper = ({ store, children, initialEntries = ['/insights/dashboard'] }) => (
  <MemoryRouter initialEntries={initialEntries}>
    <Provider store={store}>{children}</Provider>
  </MemoryRouter>
);

describe('ChromeNavItem', () => {
  const axiosGetSpy = jest.spyOn(axios.default, 'get');
  axiosGetSpy.mockImplementation(() => Promise.resolve({ data: { navItems: [] } }));
  const mockStore = createMockStore();
  const navTitle = 'Nav title';
  const store = mockStore({
    chrome: {
      activeModule: 'insights',
      modules: {
        insights: {},
      },
      navigation: {
        insights: {
          title: navTitle,
          navItems: [],
        },
      },
    },
  });

  test('should render navigation loader if schema was not loaded', () => {
    const store = mockStore({
      chrome: {
        activeModule: 'insights',
        modules: {
          insights: {},
        },
        navigation: {},
      },
    });
    const { container } = render(
      <NavContextWrapper store={store}>
        <Navigation />
      </NavContextWrapper>
    );
    expect(container).toMatchSnapshot();
    expect(container.querySelectorAll('.chr-c-app-switcher--loading')).toHaveLength(1);
    expect(container.querySelectorAll('.chr-c-app-title')).toHaveLength(0);
  });

  test('should render navigation ', async () => {
    let container;
    await act(async () => {
      const { container: iContainer } = render(
        <NavContextWrapper store={store}>
          <Navigation />
        </NavContextWrapper>
      );
      container = iContainer;
    });
    expect(container).toMatchSnapshot();
    expect(container.querySelectorAll('.chr-c-app-switcher--loading')).toHaveLength(0);
    expect(container.querySelectorAll('.chr-c-app-title')).toHaveLength(1);
    expect(container.querySelector('.chr-c-app-title').textContent).toEqual(navTitle);
  });
});
