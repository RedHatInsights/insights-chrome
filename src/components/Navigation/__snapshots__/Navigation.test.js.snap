// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChromeNavItem should render navigation  1`] = `
<div>
  <div
    class="chr-c-app-title"
  >
    Nav title
  </div>
  <nav
    aria-label="Insights Global Navigation"
    class="pf-c-nav pf-m-scrollable"
    data-ouia-component-id="SideNavigation"
    data-ouia-component-type="PF4/Nav"
    data-ouia-safe="true"
  >
    <ul
      class="pf-c-nav__list"
    />
  </nav>
</div>
`;

exports[`ChromeNavItem should render navigation loader if schema was not loaded 1`] = `
<div>
  <section
    class="chr-c-app-switcher--loading"
  >
    <div
      class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__lg ins-m-dark"
    >
      <span
        class="pf-u-screen-reader"
      />
    </div>
  </section>
  <nav
    aria-label="Insights Global Navigation"
    class="pf-c-nav"
    data-ouia-component-id="SideNavigation"
    data-ouia-component-type="PF4/Nav"
    data-ouia-safe="false"
  >
    <ul
      class="pf-c-nav__list"
    >
      <li
        class="pf-c-nav__item"
        data-ouia-component-id="OUIA-Generated-NavItem-1"
        data-ouia-component-type="PF4/NavItem"
        data-ouia-safe="true"
      >
        <a
          class="pf-c-nav__link"
          href="#"
        >
          <div
            class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__lg ins-m-dark"
          >
            <span
              class="pf-u-screen-reader"
            />
          </div>
        </a>
      </li>
      <li
        class="pf-c-nav__item"
        data-ouia-component-id="OUIA-Generated-NavItem-2"
        data-ouia-component-type="PF4/NavItem"
        data-ouia-safe="true"
      >
        <a
          class="pf-c-nav__link"
          href="#"
        >
          <div
            class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__lg ins-m-dark"
          >
            <span
              class="pf-u-screen-reader"
            />
          </div>
        </a>
      </li>
      <li
        class="pf-c-nav__item"
        data-ouia-component-id="OUIA-Generated-NavItem-3"
        data-ouia-component-type="PF4/NavItem"
        data-ouia-safe="true"
      >
        <a
          class="pf-c-nav__link"
          href="#"
        >
          <div
            class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__lg ins-m-dark"
          >
            <span
              class="pf-u-screen-reader"
            />
          </div>
        </a>
      </li>
      <li
        class="pf-c-nav__item"
        data-ouia-component-id="OUIA-Generated-NavItem-4"
        data-ouia-component-type="PF4/NavItem"
        data-ouia-safe="true"
      >
        <a
          class="pf-c-nav__link"
          href="#"
        >
          <div
            class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__lg ins-m-dark"
          >
            <span
              class="pf-u-screen-reader"
            />
          </div>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
